capybara/app_test.rb

```diff
@@ -0,0 +1,146 @@
+    @current_page = Home.new.navigate_to_login_page
+    @current_page = @current_page.log_in(CORRECT_EMAIL, CORRECT_PASS, true)
+    @current_page = @current_page.logout
+    if @current_page.user_logged_in?
+      AppLogger.log.info('[FAIL] User is expected to be logged out')
+    else
+      AppLogger.log.info('[PASS] User is logged out')
+    end
+  end
+
+  def blank_password_test
+    AppLogger.log.info('TC: User can not login with blank password')
+    @current_page = Home.new.navigate_to_login_page
+    @current_page = @current_page.log_in(CORRECT_EMAIL, '', true)
+    if @current_page.login_failed?
+      AppLogger.log.info('[PASS] User is not logged in and message: '\
+                         '"Invalid email or password" appeared')
+    else
+      AppLogger.log.info('[FAIL] User is logged in')
+    end
+  end
+
+  def blank_email_test
+    AppLogger.log.info('TC: User can not login with blank email')
+    @current_page = Home.new.navigate_to_login_page
+    @current_page = @current_page.log_in('', CORRECT_PASS, true)
+    if @current_page.login_failed?
+      AppLogger.log.info('[PASS] User is not logged in and message: '\
+                         '"Invalid email or password" appeared')
+    else
+      AppLogger.log.info('[FAIL] User is logged in')
+    end
+  end
+
+  def blank_data_test
+    AppLogger.log.info('TC: User can not login with blank fields')
+    @current_page = Home.new.navigate_to_login_page
+    @current_page = @current_page.log_in('', '', true)
+    if @current_page.login_failed?
+      AppLogger.log.info('[PASS] User is not logged in and message: '\
+                         '"Invalid email or password" appeared')
+    else
+      AppLogger.log.info('[FAIL] User is logged in')
+    end
+  end
+
+  def incorrect_email_test
+    AppLogger.log.info('TC: User can not login with incorrect email')
+    @current_page = Home.new.navigate_to_login_page
+    @current_page = @current_page.log_in(INCORRECT_EMAIL, CORRECT_PASS, true)
+    if @current_page.login_failed?
+      AppLogger.log.info('[PASS] User is not logged in and message: '\
+                         '"Invalid email or password" appeared')
+    else
+      AppLogger.log.info('[FAIL] User is logged in')
+    end
+  end
+
+  def incorrect_password_test
+    AppLogger.log.info('TC: User can not login with incorrect password')
+    @current_page = Home.new.navigate_to_login_page
+    @current_page = @current_page.log_in(CORRECT_EMAIL, INCORRECT_PASS, true)
+    if @current_page.login_failed?
+      AppLogger.log.info('[PASS] User is not logged in and message: '\
+                         '"Invalid email or password" appeared')
+    else
+      AppLogger.log.info('[FAIL] User is logged in')
+    end
+  end
+
+  def incorrect_data_test
+    AppLogger.log.info('TC: User can not login with incorrect data')
+    @current_page = Home.new.navigate_to_login_page
+    @current_page = @current_page.log_in(INCORRECT_EMAIL, INCORRECT_PASS, true)
```

Comment:
We have a lot of instance variables here. I would suggest to use getter/setter

using current_page everywhere is annoying. I would expect, something like that:

```ruby
class Web
  def self.current_page
  end
end

class Home < Web
end

Home.new.navigate_to_login_page
LoginPage.log_in(INCORRECT_EMAIL, INCORRECT_PASS, true)
if Web.login_failed?
end
```